packages/nuxi/src/dev/utils.ts

```diff
@@ -340,10 +340,12 @@ export class NuxtDevServer extends EventEmitter<DevServerEventMap> {
     if (!process.env.NUXI_DISABLE_VITE_HMR) {
       this.#currentNuxt.hooks.hook('vite:extend', ({ config }) => {
         if (config.server) {
+          // Preserve user-defined HMR port to prevent conflicts in monorepo setup
+          const existingPort = typeof config.server.hmr === 'object' ? config.server.hmr.port : undefined
           config.server.hmr = {
             protocol: undefined,
             ...(config.server.hmr as Exclude<typeof config.server.hmr, boolean>),
-            port: undefined,
+            port: existingPort,
```

Comment:
wouldn't it have the same effect simply to delete this line?

Reply:
Maybe, Right now my head is completely confused, I'm rethinking everything. I hope I can get it working one last time.

Reply:
Yes thats true